disco_aws_automation/disco_deploy.py

```diff
@@ -625,9 +625,9 @@ def _generate_post_deploy_pipeline(self, pipeline_dict, old_group, ami):
         # If there is an already existing ASG, use its sizing. Otherwise, use the pipeline's sizing or a
         # reasonable default.
         if old_group:
-            desired_size = old_group.desired_capacity
-            max_size = old_group.max_size
-            min_size = old_group.min_size
+            desired_size = old_group.get('desired_capacity') or old_group['capacity']['target']
+            max_size = old_group.get('max_size') or old_group['capacity']['maximum']
```

Comment:
ASG and Spotinst groups dictionary returned are different: value `max_size` exists only for ASG group dict, and `old_group.get('max_size')` will return `None` if group type is Spotinst, and as a result will assign value from `old_group['capacity']['maximum']`.

If we match the returned value of `get_existing_group` method for both ASG and Spotinst, as you suggested above, then this part will be simpler and less *weird*.